core/backend/src/BlobContainerService.ts

```diff
@@ -87,21 +87,29 @@ export namespace BlobContainer {
   /** The URI and Id of the container. */
   export interface UriAndId {
     baseUri: string;
-    id: ContainerId;
+    containerId: ContainerId;
   }
 
   /** Information required to access an existing container. */
-  export interface AccessContainerProps {
-    address: UriAndId;
+  export interface AccessContainerProps extends UriAndId {
     userToken: UserToken;
   }
 
+  /**
+   * Access level to request for token.
+   * - `"write"`: request a writeable AccessToken.
+   * - `"read"`:  request a read-only AccessToken.
+   * - `"admin"`:  request a admin AccessToken.
+   * - `"writeIfPossible"`: first request a writeable AccessToken. If the user is not authorized for write, request a read-only AccessToken.
+   * If the user is not authorized for the level requested, an exception is thrown.
+   */
+  export type RequestAccessLevel = "write" | "read" | "admin" | "writeIfPossible";
+
   /** Information required to request an access token for a container. */
   export interface RequestTokenProps extends AccessContainerProps {
-    /** If true, token should provide write access.
-     * @note if write access is requested and the user is authorized for read but not write, an exception will be thrown (i.e. a read token is *not* returned).
-     */
-    forWriteAccess?: boolean;
+    storageType: Provider;
```

Comment:
Why would we want to let a client choose this?

I'd prefer to just say the service always dictates the provider - that gives us a lot more flexibility and control over our costs...

Reply:
This is for the request for an access token. This was returned when the container was created, so you have to supply it to get an access token. The choice of supplier is made in the create-new-container call. After that the container can’t change storage type, right?